aten/src/ATen/native/quantized/affine_quantizer.cpp

```diff
@@ -0,0 +1,315 @@
+#include <ATen/native/quantized/affine_quantizer.h>
+
+#ifdef USE_FBGEMM
+#include <fbgemm/QuantUtils.h>
+#endif
+#ifdef __ARM_NEON__
+#include <arm_neon.h>
+#endif
+
+namespace at {
+namespace native {
+
+DEFINE_DISPATCH(quantize_tensor_affine_stub);
+DEFINE_DISPATCH(quantize_tensor_per_channel_affine_stub);
+DEFINE_DISPATCH(dequantize_tensor_affine_stub);
+DEFINE_DISPATCH(dequantize_tensor_per_channel_affine_stub);
+
+namespace {
+
+void checkCPUTensor(std::string fn_name, Tensor t) {
+  TORCH_CHECK(
+      t.device().type() == kCPU,
+      fn_name,
+      " expects a CPU Tensor.");
+}
+
+void checkFloatTensor(std::string fn_name, Tensor t) {
+  TORCH_CHECK(
+      t.scalar_type() == kFloat,
+      fn_name,
+      " expects a Float Tensor.");
+}
+
+void checkSameDevice(std::string fn_name, Tensor t1, Tensor t2) {
+  TORCH_CHECK(
+      t1.device() == t2.device(),
+      fn_name,
+      " expects a quantized and float tensors to be on the same device.");
+}
+
+template <typename T>
+void checkQuantizedTensor(std::string fn_name, Tensor t) {
+  TORCH_CHECK(t.is_quantized(),
+           fn_name,
+           " expects a quantized Tensor.");
+  TORCH_CHECK(t.scalar_type() == caffe2::TypeMeta::Make<T>(),
+           fn_name,
+           " expects a ",
+           caffe2::TypeMeta::Make<T>(),
+           " Tensor");
+  TORCH_CHECK(t.is_cuda() || t.device().type() == kCPU,
```

Comment:
Wouldn't that always be true?

Reply:
Indeed, we might omit this check for now. Fixed